### PR TITLE
 mediagoblin: init at 0.14.0,  nixos/mediagoblin: init 

### PR DESCRIPTION
**Pull request by SuperSandro2000**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: nixos/modules/services/web-apps/mediagoblin.nix [hunk 1]
```diff
@@ -0,0 +1,322 @@
...
+
+  mkSubSectionKeyValue = depth: k: v: if lib.isAttrs v then let
+    inherit (lib.strings) replicate;
+  in (replicate depth "[") + k + (replicate depth "]") + "\n" + lib.generators.toINIWithGlobalSection {
+      mkKeyValue = mkSubSectionKeyValue (depth+1);
+    } { globalSection = v; }
+  else
+    lib.generators.mkKeyValueDefault {
+      mkValueString = v: if lib.isString v then ''"${v}"''
+      else lib.generators.mkValueStringDefault { } v;
```
[the remark this answers is not in the document]
Comment (SuperSandro2000) [reply nixos/modules/services/web-apps/mediagoblin.nix:14]: That's just me hacking things together over multiple days 😅 

and yeah, I forgot to run the uglifier.

File: pkgs/development/python-modules/wtforms-sqlalchemy/default.nix [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+python3.pkgs.buildPythonPackage rec {
+  pname = "wtforms-sqlalchemy";
+  version = "0.4.1";
+  pyproject = true;
+
+  src = fetchFromGitHub {
+    owner = "wtforms";
+    repo = "wtforms-sqlalchemy";
+    rev = version;
```
Comment (GaetanLepage) [inline pkgs/development/python-modules/wtforms-sqlalchemy/default.nix:19]:
```suggestion
    rev = "refs/tags/${version}";
```

File: pkgs/development/python-modules/wtforms-sqlalchemy/default.nix [hunk 3]
```diff
@@ -0,0 +1,45 @@
+{
+  lib,
+  python3,
```
Comment (GaetanLepage) [inline pkgs/development/python-modules/wtforms-sqlalchemy/default.nix:3]:
```suggestion
  python3Packages,
```

File: pkgs/development/python-modules/wtforms-sqlalchemy/default.nix [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+  fetchFromGitHub,
+  pytestCheckHook,
+  setuptools,
+  sqlalchemy,
+  wtforms,
+}:
+
+python3.pkgs.buildPythonPackage rec {
```
Comment (GaetanLepage) [inline pkgs/development/python-modules/wtforms-sqlalchemy/default.nix:11]:
```suggestion
python3Packages.buildPythonPackage rec {
```

File: pkgs/development/python-modules/wtforms-sqlalchemy/default.nix [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+    wtforms
+  ];
+
+  nativeCheckInputs = [
+    pytestCheckHook
+  ];
+
+  pythonImportsCheck = [ "wtforms_sqlalchemy" ];
+
+  meta = with lib; {
```
Comment (GaetanLepage) [inline pkgs/development/python-modules/wtforms-sqlalchemy/default.nix:38]:
```suggestion
  meta = {
```

File: pkgs/development/python-modules/wtforms-sqlalchemy/default.nix [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+    description = "WTForms integration for SQLAlchemy";
+    homepage = "https://github.com/wtforms/wtforms-sqlalchemy";
+    changelog = "https://github.com/wtforms/wtforms-sqlalchemy/blob/${src.rev}/CHANGES.rst";
```
Comment (GaetanLepage) [inline pkgs/development/python-modules/wtforms-sqlalchemy/default.nix:41]:
```suggestion
    changelog = "https://github.com/wtforms/wtforms-sqlalchemy/blob/${lib.removePrefix "refs/tags/" src.rev}/CHANGES.rst";
```

File: pkgs/by-name/me/mediagoblin/package.nix [hunk 7]
```diff
@@ -0,0 +1,156 @@
...
+    ]
+    ++ lib.flatten (lib.attrValues optional-dependencies);
+
+  pythonImportsCheck = [ "mediagoblin" ];
+
+  passthru = {
+    inherit python;
+  };
+
+  meta = with lib; {
```
Comment (GaetanLepage) [inline pkgs/by-name/me/mediagoblin/package.nix:150]:
```suggestion
  meta = {
```

**Review by GaetanLepage — CHANGES_REQUESTED** [review]
First trivial comments.

File: pkgs/development/python-modules/wtforms-sqlalchemy/default.nix (hunk 6 above)
Comment (SuperSandro2000) [reply pkgs/development/python-modules/wtforms-sqlalchemy/default.nix:41]: we can just use version here, that is a lot easier

**Comment by GaetanLepage** [comment]
Sorry for the nitpicks, feel free to ignore them.

I have an issue with downloading the source of `mediagoblin` though:
```
fatal: unable to access 'https://git.sr.ht/~mediagoblin/mediagoblin/': The requested URL returned error: 500
```

**Comment by GaetanLepage** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 350578`

---
### `x86_64-linux`
<details>
  <summary>:fast_forward: 1 package blacklisted:</summary>
  <ul>
    <li>nixos-install-tools</li>
  </ul>
</details>
<details>
  <summary>:x: 2 packages failed to build:</summary>
  <ul>
    <li>mediagoblin</li>
    <li>mediagoblin.dist</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 4 packages built:</summary>
  <ul>
    <li>python311Packages.wtforms-sqlalchemy</li>
    <li>python311Packages.wtforms-sqlalchemy.dist</li>
    <li>python312Packages.wtforms-sqlalchemy</li>
    <li>python312Packages.wtforms-sqlalchemy.dist</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:fast_forward: 1 package blacklisted:</summary>
  <ul>
    <li>nixos-install-tools</li>
  </ul>
</details>
<details>
  <summary>:x: 2 packages failed to build:</summary>
  <ul>
    <li>mediagoblin</li>
    <li>mediagoblin.dist</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 4 packages built:</summary>
  <ul>
    <li>python311Packages.wtforms-sqlalchemy</li>
    <li>python311Packages.wtforms-sqlalchemy.dist</li>
    <li>python312Packages.wtforms-sqlalchemy</li>
    <li>python312Packages.wtforms-sqlalchemy.dist</li>
  </ul>
</details>

---
### `x86_64-darwin`
<details>
  <summary>:fast_forward: 2 packages marked as broken and skipped:</summary>
  <ul>
    <li>mediagoblin</li>
    <li>mediagoblin.dist</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 4 packages built:</summary>
  <ul>
    <li>python311Packages.wtforms-sqlalchemy</li>
    <li>python311Packages.wtforms-sqlalchemy.dist</li>
    <li>python312Packages.wtforms-sqlalchemy</li>
    <li>python312Packages.wtforms-sqlalchemy.dist</li>
  </ul>
</details>

---
### `aarch64-darwin`
<details>
  <summary>:fast_forward: 2 packages marked as broken and skipped:</summary>
  <ul>
    <li>mediagoblin</li>
    <li>mediagoblin.dist</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 4 packages built:</summary>
  <ul>
    <li>python311Packages.wtforms-sqlalchemy</li>
    <li>python311Packages.wtforms-sqlalchemy.dist</li>
    <li>python312Packages.wtforms-sqlalchemy</li>
    <li>python312Packages.wtforms-sqlalchemy.dist</li>
  </ul>
</details>

**Comment by SuperSandro2000** [comment]
> I have an issue with downloading the source of `mediagoblin` though:

sourcehut seems to be under a bit of load right now. Worked for me though.

**Comment by SuperSandro2000** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 350578`

---
### `x86_64-linux`
<details>
  <summary>:fast_forward: 1 package blacklisted:</summary>
  <ul>
    <li>nixos-install-tools</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 6 packages built:</summary>
  <ul>
    <li>mediagoblin</li>
    <li>mediagoblin.dist</li>
    <li>python311Packages.wtforms-sqlalchemy</li>
    <li>python311Packages.wtforms-sqlalchemy.dist</li>
    <li>python312Packages.wtforms-sqlalchemy</li>
    <li>python312Packages.wtforms-sqlalchemy.dist</li>
  </ul>
</details>

**Review by GaetanLepage — APPROVED** [review]
LGTM

**Pull request merged**